src/commandline.cpp

```diff
@@ -833,6 +834,51 @@ std::optional<int> ReloadPluginCommand::runPostOrganizer(OrganizerCore& core)
   return {};
 }
 
+Command::Meta DownloadFileCommand::meta() const
+{
+  return {"download", "downloads a file", "URL", ""};
+}
+
+po::options_description DownloadFileCommand::getInternalOptions() const
+{
+  po::options_description d;
+
+  d.add_options()("URL", po::value<std::string>()->required(), "file URL");
+
+  return d;
+}
+
+po::positional_options_description DownloadFileCommand::getPositional() const
+{
+  po::positional_options_description d;
+
+  d.add("URL", 1);
+
+  return d;
+}
+
+bool DownloadFileCommand::canForwardToPrimary() const
+{
+  return true;
+}
+
+std::optional<int> DownloadFileCommand::runPostOrganizer(OrganizerCore& core)
+{
+  const QString url = QString::fromStdString(vm()["URL"].as<std::string>());
+
+  if (!url.startsWith("https://")) {
+    reportError(QObject::tr("Download URL must start with https://"));
+    return 1;
+  }
+
+  log::debug("starting direct download from command line: {}", url.toStdString());
+  MessageDialog::showMessage(QObject::tr("Download started"), qApp->activeWindow(),
```

Comment:
Does this open a modal Window everytime a download is started? Isn't that a bit too much? I think a non-debug log would be enough here.

Reply:
This is not a modal window command. It shows that small tooltip above task bar for a couple second with the text, same way Nexus downloads start.